### ISSUE TITLE
`input.uploadFile` returns instantly without uploading files

### ISSUE THREAD

**Issue by lance-swoop** [puppeteer/puppeteer#5435]
Here is my simple implementation (given I am logged in properly, and every other form of puppeteer works pretty much):

```js
async function uploadFiles(p, selector, ...files) {
  const inputUploadHandle = await p.$(selector)
  await inputUploadHandle.uploadFile(...files)
}
```

I call it like:

```js
console.log('a')
await uploadFiles(p,  'input[type="file"]', '/foo/1.txt', '/foo/2.txt')
console.log('b')
```

The console logs `a` then immediately `b`. When I run in visible browser mode (not headless), I check the network as it's running and it's not making any network POST requests like it does if I manually try to click the element and select the file. Running in headless mode I have traced the requests and I get the same result that I see in visible browser mode: no requests resembling a file upload request like I see when manually uploading a file.

I have tried:

- uploading one file instead of many.
- making sure the file actually exists in this absolute path (which it does) and it has content (which it does).
- waiting for 50000ms after the file upload call to see if anything happens, but nope.
- ...

Any ideas what is happening? Could there be a hidden element actually somewhere else that, when I click on the `input[type="file"]` in the DOM, it redirects to a click on a hidden `input[type="file"]`? And so trying to upload using this input is useless? Or what could be happening?

Maybe [iframes](https://stackoverflow.com/questions/33732655/html-file-upload-why-use-iframe) are involved in the actual uploading as a redirect from the "fake" file input?

**Comment by lance-swoop** [comment]
So it looks like onclick on the file input is redirected to an XHR request, so how can I do a file upload using the URL I see actually invoked using this XHR request? How do I attach files to the request?

**Comment by paullewis** [comment]
Calling `uploadFiles` effectively "uploads" the file contents to the file input element, it won't trigger a form submission. It will depend on the specific implementation of the form in question, but you will want to use `uploadFiles` as you have (though we have just fixed its behavior in 532ae573d29063ffd081ed9ee340d71b25320dec, so you may need to be running from master) and then either submit the form, or click the button which triggers the XHR.

Closing this bug because it's WAI.
